packages/experiments/src/components/Chiclet/Chiclet.types.ts

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { ChicletBase } from './Chiclet.base';
+import { IStyleFunction } from '../../Utilities';
+import {
+  IStyle,
+  ITheme
+} from '../../Styling';
+
+export interface IChiclet {
+
+}
+
+export interface IChicletProps extends React.Props<ChicletBase> {
+  /**
+   * Optional callback to access the IChiclet interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IChiclet | null) => void;
+
+  /**
+   * Call to provide customized styling that will layer on top of the variant rules.
+   */
+  getStyles?: IStyleFunction<IChicletStyleProps, IChicletStyles>;
+
+  /**
+   * Optional class for chiclet.
+   */
+  className?: string;
+
+  /**
+   * Sharing link
+   */
+  url: string;
+
+  /**
+   * Chiclet size to render
+   */
+  size?: ChicletSize;
+
+  /**
+   * Description to render for the component.
+   */
+  description?: React.ReactElement<any>
```

Comment:
most fabric components default to strings for description. is there a specific need for using any element? Makes it a bit less predictable.

Reply:
Good question! For now, the description here serves as a kind of "location" for the chiclet, so the idea is to be able to use either a url string or a Breadcrumb, which is why I've made it more generic. One example I have with the Breadcrumb is in Chiclet.Breadcrumb.Example.tsx and one with the basic url string is in Chiclet.Basic.Example.tsx.

In the future, depending on who is using the chiclet, users may want to be able to pass in their own "description" here. Perhaps they'll just want text, or maybe something a little richer like the breadcrumb, or maybe something of their own.

@aditima may be able to speak a little more to this idea.